string problem_def
---
string[] plan
